pkg/chunk/azure/blob_storage_client.go

```diff
@@ -17,11 +17,37 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/flagext"
 )
 
-const blobURLFmt = "https://%s.blob.core.windows.net/%s/%s"
-const containerURLFmt = "https://%s.blob.core.windows.net/%s"
+const (
+	//Blob service endpoint of AzureGlobal
+	globalBlobURLFmt      = "https://%s.blob.core.windows.net/%s/%s"
+	globalContainerURLFmt = "https://%s.blob.core.windows.net/%s"
+
+	// Blob service endpoint of AzureChinaCloud
+	chinaBlobURLFmt      = "https://%s.blob.core.chinacloudapi.cn/%s/%s"
+	chinaContainerURLFmt = "https://%s.blob.core.chinacloudapi.cn/%s"
+
+	// Blob service endpoint of AzureGermanCloud
+	germanyBlobURLFmt      = "https://%s.blob.core.cloudapi.de/%s/%s"
+	germanyContainerURLFmt = "https://%s.blob.core.cloudapi.de/%s"
+
+	// Blob service endpoint of AzureUSGovernment
+	govBlobURLFmt      = "https://%s.blob.core.usgovcloudapi.net/%s/%s"
+	govContainerURLFmt = "https://%s.blob.core.usgovcloudapi.net/%s"
```

Comment:
I'm trying to verify these endpoints, but cannot find authoritative summary documentation. Do you have any relevant link?

Would it make sense to use something like:

```go
var endpoints = map[string]struct{blobURL, containerURL string} {
    azureGlobal: {
        blobURL: "https://%s.blob.core.windows.net/%s/%s",
        containerURL: "https://%s.blob.core.windows.net/%s",
    },

    azureChinaCloud: {
        blobURL: "https://%s.blob.core.chinacloudapi.cn/%s/%s",
        containerURL: "https://%s.blob.core.chinacloudapi.cn/%s",
    },

    // ...
}
```

and then use the map instead of `selectBlobURLFmt` and `selectContainerURLFmt`?

Reply:
Please refer to the following document：

> **China:** https://docs.microsoft.com/en-us/azure/china/resources-developer-guide
**Germany:** https://docs.microsoft.com/ca-es/azure////germany/germany-services-storage
**US Government:** https://docs.microsoft.com/en-us/azure/azure-government/compare-azure-government-global-azure

Reply:
> I'm trying to verify these endpoints, but cannot find authoritative summary documentation. Do you have any relevant link?
> 
> Would it make sense to use something like:
> 
> ```go
> var endpoints = map[string]struct{blobURL, containerURL string} {
>     azureGlobal: {
>         blobURL: "https://%s.blob.core.windows.net/%s/%s",
>         containerURL: "https://%s.blob.core.windows.net/%s",
>     },
> 
>     azureChinaCloud: {
>         blobURL: "https://%s.blob.core.chinacloudapi.cn/%s/%s",
>         containerURL: "https://%s.blob.core.chinacloudapi.cn/%s",
>     },
> 
>     // ...
> }
> ```
> 
> and then use the map instead of `selectBlobURLFmt` and `selectContainerURLFmt`?

done